src/test/java/com/smartbear/readyapi/client/SoapRequestStepRecipeTest.java

```diff
@@ -109,4 +112,56 @@ private void assertRequestAttachment(RequestAttachment attachment, String conten
         assertThat(attachment.getName(), is(name));
         assertThat(Base64.decode(attachment.getContent()), is(content));
     }
+
+    @Test
+    public void buildRestRequestTestRecipeWithPropertyExtractor() throws MalformedURLException {
+        final String[] extractedProperty = {""};
+        TestRecipe recipe = newTestRecipe()
+                .addStep(soapRequest(new URL("http://www.webservicex.com/globalweather.asmx?WSDL"))
+                        .named("Soap Rulez")
+                        .forBinding("GlobalWeatherSoap12")
+                        .forOperation("GetWeather")
+                        .withParameter("CountryName", "Sweden")
+                        .withPathParameter("//*:CityName", "Stockholm")
+                        .withExtractors(
+                                extractorProperty("Endpoint", property -> extractedProperty[0] = property)))
+                .buildTestRecipe();
+
+        // This should not be set only after building the testrecipe, it should be set after run
+        assertThat(extractedProperty[0], is(""));
+        assertThat(recipe.getTestCase().getProperties().size(), is(2));
+        recipe.getTestCase().getProperties().forEach((key, value) -> {
+            if(key.contains("Endpoint")) {
+                assertThat(value, is(""));
+            } else {
+                assertThat(key, is(ExtractorData.EXTRACTOR_DATA_KEY));
+            }
+        });
+    }
+
+    @Test
+    public void buildRestRequestTestRecipeWithJsonPathExtractor() throws MalformedURLException {
+        final String[] extractedProperty = {""};
+        TestRecipe recipe = newTestRecipe()
+                .addStep(soapRequest(new URL("http://www.webservicex.com/globalweather.asmx?WSDL"))
+                        .named("Soap Rulez")
+                        .forBinding("GlobalWeatherSoap12")
+                        .forOperation("GetWeather")
+                        .withParameter("CountryName", "Sweden")
+                        .withPathParameter("//*:CityName", "Stockholm")
+                        .withExtractors(
+                                extractorPath("$[0].Endpoint", property -> extractedProperty[0] = property)))
```

Comment:
could we add a pathExtractor that actually extracts a value from the XML response?